WalletWasabi.Fluent/ViewModels/SearchBar/SearchBarViewModel.cs

```diff
@@ -37,13 +37,13 @@ public SearchBarViewModel(IObservable<IChangeSet<ISearchItem, ComposedKey>> item
 			.ObserveOn(RxApp.MainThreadScheduler)
 			.Subscribe();
 
-		Any = filteredItems
+		_hasResults = filteredItems
 			.Count()
 			.Select(i => i > 0)
-			.ObserveOn(RxApp.MainThreadScheduler);
+			.ToProperty(this, x => x.HasResults);
 	}
 
-	public IObservable<bool> Any { get; }
+	public bool HasResults => _hasResults.Value;
```

Comment:
Yes. This fixes a bug: 
> 2. Invalid state when filter shows no results after focus regain.

I think Avalonia has a problem binding to observables. To overcome the issue we either need **OAPH** or a **ReplayBehavior**. I chose the former because this is how it's done in RxUI.

NOTE: I'm still investigating the issue binding to observables.